src/websocket/websocketconnection.c

```diff
@@ -430,9 +430,10 @@ send_prefixed_message_rfc6455 (WebSocketConnection *self,
    */
   if (!self->pv->server_side)
     {
+      guint32 rand = g_random_int ();
       outer[1] |= 0x80;
       mask = outer + bytes->len;
-      * ((guint32 *)mask) = g_random_int ();
+      memcpy (mask, &rand , sizeof (guint8));
```

Comment:
Thanks @glaubitz for tracking that down! But I wonder, why does this only copy the first 8 bits instead of all 32? Shouldn't it be `sizeof guint32`?

Reply:
The target variable ```mask``` is only 8 bits (```guint8```, see at the top the function), so copying 32 bits would result in memory being overwritten.

Reply:
Hmm, although now I'm a bit confused why it worked on the other architectures in the first place without crashing when a 32-bit value is written to a 8-bit destination.